From ceb2c936fd3a93e00db4b90fff00ac8a4d0f2e68 Mon Sep 17 00:00:00 2001
From: Brett Mastbergen <bmastbergen@ciq.com>
Date: Fri, 22 Aug 2025 12:21:36 -0400
Subject: [PATCH] net: atm: fix use after free in lec_send()

jira VULN-56261
cve CVE-2025-22004
commit-author Dan Carpenter <dan.carpenter@linaro.org>
commit f3009d0d6ab78053117f8857b921a8237f4d17b3

The ->send() operation frees skb so save the length before calling
->send() to avoid a use after free.

Fixes: 1da177e4c3f4 ("Linux-2.6.12-rc2")
	Signed-off-by: Dan Carpenter <dan.carpenter@linaro.org>
	Reviewed-by: Simon Horman <horms@kernel.org>
Link: https://patch.msgid.link/c751531d-4af4-42fe-affe-6104b34b791d@stanley.mountain
	Signed-off-by: Paolo Abeni <pabeni@redhat.com>

(cherry picked from commit f3009d0d6ab78053117f8857b921a8237f4d17b3)
	Signed-off-by: Brett Mastbergen <bmastbergen@ciq.com>
---
 net/atm/lec.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/net/atm/lec.c b/net/atm/lec.c
index 21b0caf3aad0..f36eb5fc734c 100644
--- a/net/atm/lec.c
+++ b/net/atm/lec.c
@@ -180,6 +180,7 @@ static void
 lec_send(struct atm_vcc *vcc, struct sk_buff *skb)
 {
 	struct net_device *dev = skb->dev;
+	unsigned int len = skb->len;
 
 	ATM_SKB(skb)->vcc = vcc;
 	ATM_SKB(skb)->atm_options = vcc->atm_options;
@@ -191,7 +192,7 @@ lec_send(struct atm_vcc *vcc, struct sk_buff *skb)
 	}
 
 	dev->stats.tx_packets++;
-	dev->stats.tx_bytes += skb->len;
+	dev->stats.tx_bytes += len;
 }
 
 static void lec_tx_timeout(struct net_device *dev)
-- 
2.47.3

